src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/LightLambda.cs

```diff
@@ -377,9 +380,47 @@ internal Delegate MakeDelegate(Type delegateType)
 #endif
         }
 
+        private InterpretedFrame? GetCachedFrame()
```

Comment:
Can `InterpretedFrame` live on the stack instead?

Reply:
Making `InterpretedFrame` would be attractive, but involved. We pass it around a lot - basically to every instruction `Run` method. That would need to be passed byref and ensure there are no accidental copying of the state. 

Also frames form a chain - there is a Parent frame and there is `s_currentFrame` which is `ThreadStatic`. 
Perhaps all uses that are fields could be changed to be native pointers.

Reply:
I have made `InterpretedFrame` a struct and pass it around by reference. It is a big change, but fairly mechanical (99% search replace).

The eval stack still needs to be Rented/Returned since it is of variable size, but for this I can use `ArrayPool<object>.Shared`, since we do not need exact length. 
Same for rarely needed `_continuations` array.

The only ugly-ish part is chaining the frames. The `_parent` and `s_currentFrame` are pointers now, which should work ok, since frames that are pointed there are always on stack.

Reply:
The perf is better with stack-based frames. I think it is because allocation of `InterpreterFrame` is completely gone and renting of the eval stack is more effective.

After this change we get to ~630.000 RPS in the BasicMinimalApi, with baseline ~600.000 RPS.  roughly ~5% improvement.